### PR TITLE
Device meta

### PR DESCRIPTION
**Pull request by willmcgugan**
Adds support for new meta methods.

### CODE REVIEW

File: dataplicity/client/client.py [hunk 1]
```diff
@@ -477,7 +508,21 @@ def _sync(self):
             try:
                 batch.get_result('set_firmware_result')
             except Exception as e:
-                self.log.warning("unable to set firmware ({})".format(e))
+                self.log.warning("unable to set firmware (%s)", e)
+
+            if not self._sent_meta:
```
Comment (willmcgugan) [inline dataplicity/client/client.py]: Check the api calls worked. If they did, we don't need to call them again next sync.

File: dataplicity/client/device_meta.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    """Get uname."""
+    # Preferable to running a system command
+    uname = ' '.join(platform.uname())
+    return uname
+
+
+def get_os_version():
+    """Get the OS version."""
+    # Linux is a fair assumption for now
+    distro = " ".join(platform.linux_distribution()).strip()
```
Comment (willmcgugan) [inline dataplicity/client/device_meta.py]: Apparently this is deprecated. But we have a few years at least to improve it.

File: dataplicity/jsonrpc.py [hunk 3]
```diff
@@ -151,6 +151,11 @@ def get_result(self, call_id, default=Ellipsis):
         else:
             raise KeyError("No such call_id in response")
 
+    def check(self, *call_ids):
```
Comment (willmcgugan) [inline dataplicity/jsonrpc.py]: New method to raise any exceptions but discard return value(s).

File: dataplicity/rpi.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
+from __future__ import unicode_literals
+
+RPI_REVISIONS = {
```
Comment (willmcgugan) [inline dataplicity/rpi.py]: Update from http://www.raspberrypi-spy.co.uk/2012/09/checking-your-raspberry-pi-board-version/

**Comment by willmcgugan** [comment]
@wildfoundry/web-developers Please review

This needs to be pushed to PyPi prior to TTNG.

**Comment by meshy** [comment]
:+1:

**Comment by barfle** [comment]
:+1:

**Pull request merged**